core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/NetworkV1ToNetworkV4Converter.java

```diff
@@ -157,8 +175,23 @@ private AwsNetworkV4Parameters convertToAwsStackRequest(Pair<AwsNetworkV1Paramet
             String subnetId = key.getSubnetId();
             if (!Strings.isNullOrEmpty(subnetId)) {
                 response.setSubnetId(key.getSubnetId());
-            } else if (source.getValue() != null) {
-                response.setSubnetId(source.getValue().getPreferedSubnetId());
+            } else if (value != null) {
+                response.setSubnetId(value.getPreferedSubnetId());
+            }
+
+            if (entitlementService.publicEndpointAccessGatewayEnabled(ThreadBasedUserCrnProvider.getAccountId())) {
+                ValidationResult validationResult = endpointGatewayNetworkValidator.validate(new ImmutablePair<>(response.getSubnetId(), value));
+                if (validationResult.getState() == ValidationResult.State.ERROR || validationResult.hasError()) {
+                    throw new BadRequestException("Endpoint gateway subnet validation failed: " + validationResult.getFormattedErrors());
+                }
+
+                if (PublicEndpointAccessGateway.ENABLED.equals(value.getPublicEndpointAccessGateway())) {
+                    Optional<CloudSubnet> endpointGatewaySubnet = subnetSelector.chooseSubnetForEndpointGateway(value, response.getSubnetId());
+                    if (endpointGatewaySubnet.isPresent()) {
+                        // This should always be true because the endpointGatewayNetworkValidator verifies this optional is set
```

Comment:
Maybe throw an UnexpectedException() or something similar. Or maybe add an `assert false` statement in else?